### PR TITLE
huion-tablet: Add Cask

### PR DESCRIPTION
**Pull request by 0az**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` reports no offenses.
- [X] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask-drivers/pulls) for the same update.
- [X] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [X] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [X] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.
- [X] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask-drivers/search?q=is%3Aclosed&type=Issues).
- [X] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

**Comment by miccal** [comment]
I see a lot ofproduct models listed [here](https://www.huion.com/download/) -- which one is this for?

**Comment by 0az** [comment]
I'm pretty sure this works for multiple, given that there's none attached.

This was originally going to be for the Huion HS64, but I checked another couple and they were labelled similarly: didn't download and check hashes, though. I can do that once I get back to my laptop.

**Comment by miccal** [comment]
> I'm pretty sure this works for multiple, given that there's none attached.
> 
> This was originally going to be for the Huion HS64, but I checked another couple and they were labelled similarly: didn't download and check hashes, though. I can do that once I get back to my laptop.

For some of them the `version` is just `14.4.5`, while for others it is `14.4.5.200611`, hence my confusion.

**Comment by 0az** [comment]
Interesting. So they are different versions.

```
hdiutil info | rg 'dmg|/Volumes'
image-path      : /private/tmp/HuionTablet_MacDriver_v14.4.5.200611.dmg
image-alias     : /private/tmp/HuionTablet_MacDriver_v14.4.5.200611.dmg
/dev/disk2s2	Apple_HFS	/Volumes/HuionTablet_MacDriver_v14.4.5.200611
image-path      : /private/tmp/HuionTablet_MacDriver_v14.4.5.dmg
image-alias     : /private/tmp/HuionTablet_MacDriver_v14.4.5.dmg
/dev/disk3s2	Apple_HFS	/Volumes/HuionTablet_MacDriver_v14.4.5.200730
```

I'll test the 200730 version later.

**Comment by 0az** [comment]
Huion support says that they're the same, but .200730 is more stable.

I'll swap the "version" to 14.4.5.200730, modify the `url`, and add a comment to the cask. The next version will have to remove the hack under `url`, but this way the correct version displays.

**Comment by miccal** [comment]
> Huion support says that they're the same, but .200730 is more stable.
> 
> I'll swap the "version" to 14.4.5.200730, modify the `url`, and add a comment to the cask. The next version will have to remove the hack under `url`, but this way the correct version displays.

Perhaps `version.major_minor_patch` would work better.

File: Casks/huion-tablet.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  url "https://driver.huion.com/driver/Mac/HuionTablet_MacDriver_v#{version.sub(/\.\d+$/, "")}.zip"
+  name "Huion Tablet"
+  desc "Driver for Huion Tablets"
+  homepage "https://huion.com/"
+
+  container nested: "HuionTablet_MacDriver_v#{version.sub(/\.\d+$/, "")}.dmg"
+
+  app "HuionTablet.app"
+
+  caveats <<~EOS
```
Comment (vitorgalvao) [inline Casks/huion-tablet.rb:15]: Either change to [`unsigned_accessibility`](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/caveats.md) or remove the `caveat`

File: Casks/huion-tablet.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+cask "huion-tablet" do
+  version "14.4.5.200730"
+  sha256 "0fb410e205849dc277cef76c63eacc8cacb33faf3fec96a66247ce7a913a504a"
+
+  # Version 14.4.5.200730 does not include the final part on Huion's website
```
Comment (vitorgalvao) [inline Casks/huion-tablet.rb:5]: Remove this comment. It doesn’t add useful information.

File: Casks/huion-tablet.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  url "https://driver.huion.com/driver/Mac/HuionTablet_MacDriver_v#{version.sub(/\.\d+$/, "")}.zip"
```
Comment (vitorgalvao) [inline Casks/huion-tablet.rb:6]: Like @miccal said, `version.major_minor_patch` is the way to go.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/huiontablet.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  sha256 "0fb410e205849dc277cef76c63eacc8cacb33faf3fec96a66247ce7a913a504a"
+
+  url "https://driver.huion.com/driver/Mac/HuionTablet_MacDriver_v#{version.major_minor_patch}.zip"
+  name "Huion Tablet"
+  desc "Driver for Huion Tablets"
+  homepage "https://huion.com/"
+
+  container nested: "HuionTablet_MacDriver_v#{version.major_minor_patch}.dmg"
+
+  app "HuionTablet.app"
```
Comment (vitorgalvao) [inline Casks/huiontablet.rb:12]: Please try adding a [`zap`](https://github.com/homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md). If you have difficulties, that won’t be an impediment to merging as is.

**Review by vitorgalvao — APPROVED** [review]

**Comment by core-code** [comment]
i am not sure if things are alright here. 

when downloading the cask the *actual* version number is:

`CFBundleShortVersionString: 14.4.5.200617`

why put '14.4.5.200730' into the version number when this is clearly a different version - especially when we don't use the last version component anyway? 


i'm sorry if this has been discussed - i read every comment in this thread - but it still doesn't make sense to my why we have the wrong version number in there.

**Comment by miccal** [comment]
> i am not sure if things are alright here.
> 
> when downloading the cask the _actual_ version number is:
> 
> `CFBundleShortVersionString: 14.4.5.200617`
> 
> why put '14.4.5.200730' into the version number when this is clearly a different version - especially when we don't use the last version component anyway?
> 
> i'm sorry if this has been discussed - i read every comment in this thread - but it still doesn't make sense to my why we have the wrong version number in there.

The `url` is wrong for version `14.4.5.200730` -- it should be `https://driver.huion.com/driver/Kamvas22/HuionTablet_MacDriver_v14.4.5.zip`. 

@0az can you shed any more light on this?

**Comment by 0az** [comment]
I'm confused as well now. I think I screwed up a copy/paste somewhere.

**Comment by core-code** [comment]
if the downloaded file is correct i propose changing the version stanza to `14.4.5` or `14.4.5.200617`

and if we are not downloading the correct file, then... more work :)

**Comment by 0az** [comment]
Yeah, after further digging, it appears that I copied the wrong version or the wrong URL. 14.4.5.200617 is the correct version, and not 200611.

**Comment by core-code** [comment]
thanks for the digging

please verify: https://github.com/Homebrew/homebrew-cask-drivers/pull/1794

**Pull request merged**